src/WebJobs.Script/Scale/FunctionsScaleManager.cs

```diff
@@ -39,13 +56,17 @@ public FunctionsScaleManager(IScaleMonitorManager monitorManager, IScaleMetricsR
         /// <returns>The scale vote.</returns>
         public virtual async Task<ScaleStatusResult> GetScaleStatusAsync(ScaleStatusContext context)
         {
-            var monitors = _monitorManager.GetMonitors();
+            var scaleMonitors = _monitorManager.GetMonitors();
+            var targetScalers = _targetScalerManager.GetTargetScalers();
+
+            Utility.GetScaleInstancesToProcess(_environment, _functionsHostingConfiguration, scaleMonitors, targetScalers,
```

Comment:
Can we make this an instance method on FunctionsScaleManager named GetScaleStatusProviders that doesn't take any inputs, and returns the scaleMonitors/targetScalers as output parameters? FunctionsScaleMonitorService can take FunctionsScaleManager as a ctor param and call the same method.

This will allow us to initialize target scalers as Enumerable.Empty<ITargetScaler>() and avoid instantiation of scalers when TBS isn't enabled.

Reply:
I do not see any benefits doing this as we have underlying interfaces in place, also it will require refactoring `FunctionsScaleMonitorServiceTests` to initialize `FunctionsScaleManager` or introducing `IFunctionsScaleManager` to mock.

Reply:
All right, then to address my primary concern of instantiating target scalers when not needed, you should remove lines 59/60, and pass the monitorManager and targetScalerManager into the method so it'll do everything internally. Also, lets move the static method to FunctionsScaleManager.